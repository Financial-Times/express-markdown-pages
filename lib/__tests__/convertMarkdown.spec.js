const fs = require('fs/promises');
const path = require('path');
const subject = require('../convertMarkdown');

const fixturePath = path.join(__dirname, './__fixtures__/No frontmatter.md');

describe('lib/convertMarkdown', () => {
	let fixture;

	beforeAll(async () => {
		fixture = await fs.readFile(fixturePath, 'utf-8');
	});

	describe('.markdownToHTML()', () => {
		it('outputs HTML markup', () => {
			const result = subject.markdownToHTML(fixture);
			expect(result).toMatchSnapshot();
		});
	});

	describe('.markdownToText()', () => {
		it('does not output any HTML markup', () => {
			const result = subject.markdownToText(fixture);

			expect(result).not.toMatch('<');
			expect(result).not.toMatch('>');
		});

		it('does not output any encoded characters', () => {
			const result = subject.markdownToText(fixture);
			expect(result).toMatch('entities & "smart quotes"');
		});
	});
});
