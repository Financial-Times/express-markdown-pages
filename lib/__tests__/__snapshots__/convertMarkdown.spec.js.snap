// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/convertMarkdown .markdownToHTML() outputs HTML markup 1`] = `
"<h1 id=\\"python-basics\\">Python basics</h1>
<p>Python is a nice language, you can add stuff. Bla bla bla.</p>
<p>Lets jump into an exercise :</p>
<hr>
<p>It’s dead simple, <code>c</code> must be the sum of <code>a</code> and <code>b</code></p>
<div data-o-component=\\"o-syntax-highlight\\">
				<pre><code class=\\"o-syntax-highlight--py\\">a = 1
b = 2</code></pre>
			</div><div data-o-component=\\"o-syntax-highlight\\">
				<pre><code class=\\"o-syntax-highlight--py\\">a = 1
b = 2
c = a + b</code></pre>
			</div><div data-o-component=\\"o-syntax-highlight\\">
				<pre><code class=\\"o-syntax-highlight--py\\">assert(c, 3)</code></pre>
			</div><hr>
<p>Some more nice content ….</p>
<p><a href=\\"http://gitbook.io\\">Cool stuff</a></p>
<p><a href=\\"./xyz/file-name\\">Link to another Markdown file</a></p>
<p>And a test of HTML entities &amp; “smart quotes”</p>
<p>And we also need to describe emoji characters like these:</p>
<ul>
<li>  <span role=\\"img\\" aria-label=\\"dog\\">🐶</span></li>
<li>  <span role=\\"img\\" aria-label=\\"smile cat\\">😸</span></li>
<li>  <span role=\\"img\\" aria-label=\\"hankey\\">💩</span></li>
</ul>
"
`;
